autode/plotting.py

```diff
@@ -9,16 +9,30 @@
 from autode.units import energy_unit_from_name
 from autode.log import logger
 
+if TYPE_CHECKING:
+    from autode.opt.optimisers.base import _OptimiserHistory
+    from matplotlib.figure import Figure
 
-def save_plot(plot, filename):
-    """Save a plot"""
+
+def save_plot(figure: "Figure", filename: str, **kwargs):
+    """
+    Save a pyplot figure
+
+    Args:
+        figure (matplotlib.figure.Figure): The matplotlib figure object
+        filename (str): Name of the file to plot
+        **kwargs : Other keyword arguments for matplotlib which
+                   are passed onto figure.savefig()
+    """
+    import matplotlib.pyplot as plt
 
     if os.path.exists(filename):
         logger.warning("Plot already exists. Overriding..")
         os.remove(filename)
 
-    plot.savefig(filename, dpi=400 if Config.high_quality_plots else 100)
-    plot.close()
+    dpi = 400 if Config.high_quality_plots else 100
+    figure.savefig(filename, dpi=dpi, **kwargs)
```

Comment:
I wonder if we ditch the `high_quality_plots` param and always use pdfs for that crispy vectorised goodness?

Reply:
@t-young31 let's leave that as it is, it can be handled in another PR?